### ISSUE TITLE
Script serialization limitation using inherance

### ISSUE THREAD

**Issue by RuanLucasGD** [FlaxEngine/FlaxEngine#1931]
**Issue description:**
You cannot serialize members of base class using Serialize and ShowInEdtitor.

![image](https://github.com/FlaxEngine/FlaxEngine/assets/79365912/7a5943d0-8c43-41b0-b757-650d0d0f9c43)

```
using FlaxEngine;

namespace Game;

public class A : Script
{
    [Serialize, ShowInEditor] private float MyVariable;
}

public class B : A
{
    public float MyVariable2;
}

```

**Flax version:**
1.7 master

**Comment by SinnersSum** [comment]
Try setting `MyVariable` to `protected` or `internal` and see if it serializes

**Comment by RuanLucasGD** [comment]
@SinnersSum using protected worked, thanks

**Comment by SinnersSum** [comment]
> @SinnersSum using protected worked, thanks.
> 
> But i need private variables yet, this would nice if have on future

Yeah I agree, would be nice to have, and should be possible to implement. I think it's a matter of changing how fields are discovered. I expect that currently flatten-hierarchy or something like it is being used to discover fields —which IIRC will skip private members from super classes—, if instead the serializer looked at each level of the hierarchy and flattened them itself it would be able to detect private fields.
